Add Textpattern

drmonkeyninja opened the pull request:
<!-- Please fill in the **bold** fields and tick all applicable boxes by placing an "x" inside "[ ]". -->

https://github.com/drmonkeyninja/awesome-textpattern

This is the awesome list for the [Textpattern CMS](http://textpattern.com) and has been around for well over a year now. The list uses Textile instead of Markdown, but this is because Textile is the default markup syntax used by the Textpattern community and so was chosen to encourage participation.
- [x] I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- [x] This pull request has a descriptive title. _(For example: `Add Node.js`)_
- The list I added
  - [x] **has been around for at least 20 days,**
  - [ ] is a non-generated Markdown file in a GitHub repo,
  - [x] is not a duplicate,
  - [x] only has awesome items with descriptions _(Awesome lists are curations of the best, not of everything)_,
  - [x] contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
  - [x] has a Table of Contents section named `Contents` as the first section,
  - [x] has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) _(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)_,
  - [x] has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
  - [x] has been checked for inconsistent formatting, spelling, and grammar _(This includes starting each link description with an uppercase character and ending it with a period `.`, and consistent naming)_,
  - [x] has been added at the bottom of the appropriate category.

davisonio commented:
> has a Table of Contents section named Contents as the first section,

Not done.

> is a non-generated Markdown file in a GitHub repo,

Any reason for not using markdown? We prefer markdown as it's much easier to use and can mean that new contributors to awesome lists can get editing right away without learning complicated syntax.

sindresorhus commented:
Don't use the en-dash between link and description. [It's not meant for that purpose](http://www.thepunctuationguide.com/en-dash.html).

I would like to see more than just plugins. Include some blog posts, tutorials, links to community resources (Example: https://github.com/sindresorhus/awesome-electron#community), videos of talks, book, learning material, etc.

drmonkeyninja commented:
@davisonio I've updated the 'Table of Contents' to just 'Contents', I'd missed that one when submitting this as I've been maintaining Awesome Textpattern for a while now and at the time the other Awesome lists all seemed to use 'Table of Contents' (and still do).

As for the use of Textile over Markdown the reasoning is basically the same as yours for using Markdown for the Awesome lists. Markdown isn't commonly used by the Textpattern community as the Textpattern CMS has always used Textile as it's main markup syntax and so is more familiar to its users. Switching the list to Markdown would add a barrier to Textpattern users to contributing to it and I'd rather make it easy for others from the CMS's community contribute to it.

@sindresorhus the en-dash is an unfortunate side effect of using Textile due to the way it gets rendered. As for non-plugin resources there are already a few listed but I've added a Community section and asked the Textpattern community to help expand this.

sindresorhus commented:
Regarding use of Textile, makes sense.

I do like to see more blog posts, tutorials, videos of talks, book, learning material, etc, being added though. Here's an example of how it's done: https://github.com/sindresorhus/awesome-electron#documentation

- I would drop the `inspired by the awesome-php idea.` part. You can rather link to that list in a `Related` section at the bottom.
- `the default theme` => `The default theme`.

sindresorhus commented on readme.md:
```diff
@@ -367,6 +367,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
 
 - [Umbraco](https://github.com/leekelleher/awesome-umbraco)
 - [Refinery CMS](https://github.com/refinerycms-contrib/awesome-refinerycms)
+- [Textpattern](https://github.com/drmonkeyninja/awesome-textpattern)
```
Can you add a succinct description about what Textpattern is here. See other description on how to format it.

drmonkeyninja commented:
@sindresorhus I've fixed the merge conflict and added a description. I've also fixed those issues with Awesome Textpattern. 😀

sindresorhus commented:
ping :)

sindresorhus commented:
Landed :)

Noticed a couple of more things now though:

- Include project logo in the top right, like e.g. https://github.com/sindresorhus/awesome-electron (The logo should link to the project website) (The logo should be high-DPI, set it to minimum half the width of the original image)
- There should be a succinct summary at the top about the project.

drmonkeyninja commented:
Thanks @sindresorhus. I'll look into getting those further points sorted. 😃
